crates/ika-move-packages/packages/ika_system/sources/dwallet/dwallet_2pc_mpc_coordinator_inner.move

```diff
@@ -1766,51 +1901,56 @@ public(package) fun is_ecdsa_presign_valid(
 /// # Aborts
 /// - **`EExtraDataAndMessagesLenMismatch`**: If `signature_algorithm_data` and `message_approvals` have different lengths.
 /// - **`EMissingApprovalOrWrongApprovalOrder`**: If message approvals are incorrect or missing.
-fun emit_ecdsa_sign_event(
+fun emit_sign_event(
     self: &mut DWalletCoordinatorInner,
     pricing: PricingPerOperation,
     payment_ika: &mut Coin<IKA>,
     payment_sui: &mut Coin<SUI>,
     message_approval: MessageApproval,
-    presign_cap: ECDSAPresignCap,
+    presign_cap: PresignCap,
     message_centralized_signature: vector<u8>,
     is_future_sign: bool,
     ctx: &mut TxContext
 ) {
     let created_at_epoch = self.current_epoch;
-    let (dwallet, public_output) = self.get_active_dwallet_and_public_output_mut(presign_cap.dwallet_id);
 
-    assert!(dwallet.ecdsa_presigns.contains(presign_cap.presign_id), EPresignNotExist);
-    let presign = dwallet.ecdsa_presigns.remove(presign_cap.presign_id);
+    assert!(self.presigns.contains(presign_cap.presign_id), EPresignNotExist);
+    let presign = self.presigns.remove(presign_cap.presign_id);
+
+    let (dwallet, dwallet_public_output) = self.get_active_dwallet_and_public_output_mut(message_approval.dwallet_id);
 
     let MessageApproval {
         dwallet_id: message_approval_dwallet_id,
+        signature_algorithm: message_approval_signature_algorithm,
         hash_scheme,
         message
     } = message_approval;
-    let ECDSAPresignCap {
+    let PresignCap {
         id,
         dwallet_id: presign_cap_dwallet_id,
         presign_id: presign_cap_presign_id,
     } = presign_cap;
     let presign_cap_id = id.to_inner();
     id.delete();
-    let ECDSAPresign {
+    let Presign {
         id,
         created_at_epoch: _,
         dwallet_id: presign_dwallet_id,
         cap_id,
         state,
+        curve,
+        signature_algorithm,
     } = presign;
     let presign = match(state) {
-        ECDSAPresignState::Completed { presign } => {
+        PresignState::Completed { presign } => {
             presign
         },
         _ => abort EInvalidPresign
     };
     let presign_id = id.to_inner();
     id.delete();
-    assert!(presign_dwallet_id == message_approval_dwallet_id, EMessageApprovalMismatch);
+    assert!(presign_dwallet_id.is_none() || presign_dwallet_id.is_some_and!(|id| id == message_approval_dwallet_id), EMessageApprovalMismatch);
+    assert!(signature_algorithm == message_approval_signature_algorithm, EMessageApprovalMismatch);
     assert!(presign_cap_id == cap_id, EPresignNotExist);
     assert!(presign_id == presign_cap_presign_id, EPresignNotExist);
     assert!(presign_cap_dwallet_id == presign_dwallet_id, EPresignNotExist);
```

Comment:
Better copy all the presign/message approval checks here too just in case: that everything is supported, that this supports keyless, nothing is paused etc